server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/TikaMailboxModule.java

```diff
@@ -0,0 +1,91 @@
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Singleton;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.tika.TikaConfiguration;
+import org.apache.james.mailbox.tika.TikaHttpClient;
+import org.apache.james.mailbox.tika.TikaHttpClientImpl;
+import org.apache.james.mailbox.tika.TikaTextExtractor;
+import org.apache.james.utils.PropertiesProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.primitives.Ints;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+
+public class TikaMailboxModule extends AbstractModule {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TikaMailboxModule.class);
+
+    private static final String TIKA_CONFIGURATION_NAME = "tika";
+    private static final String TIKA_HOST = "tika.host";
+    private static final String TIKA_PORT = "tika.port";
+    private static final String TIKA_TIMEOUT_IN_MS = "tika.timeoutInMillis";
+
+    private static final String DEFAULT_HOST = "127.0.0.1";
+    private static final int DEFAULT_PORT = 9998;
+    private static final int DEFAULT_TIMEOUT_IN_MS = Ints.checkedCast(TimeUnit.SECONDS.toMillis(30));
+
+    @Override
+    protected void configure() {
+        bind(TikaHttpClientImpl.class).in(Scopes.SINGLETON);
+        bind(TikaHttpClient.class).to(TikaHttpClientImpl.class);
+        bind(TikaTextExtractor.class).in(Scopes.SINGLETON);
+        bind(TextExtractor.class).to(TikaTextExtractor.class);
+    }
+
+    @Provides
+    @Singleton
+    protected TikaHttpClient provideTikaHttpClient(TikaConfiguration tikaConfiguration) throws URISyntaxException {
+        return new TikaHttpClientImpl(tikaConfiguration);
+    }
+
+    @Provides
+    @Singleton
+    private TikaConfiguration getTikaConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
+        try {
+            PropertiesConfiguration configuration = propertiesProvider.getConfiguration(TIKA_CONFIGURATION_NAME);
+            return TikaConfiguration.builder()
+                    .host(configuration.getString(TIKA_HOST, DEFAULT_HOST))
+                    .port(configuration.getInt(TIKA_PORT, DEFAULT_PORT))
+                    .timeoutInMillis(configuration.getInt(TIKA_TIMEOUT_IN_MS, DEFAULT_TIMEOUT_IN_MS))
+                    .build();
+        } catch (FileNotFoundException e) {
+            LOGGER.warn("Could not find " + TIKA_CONFIGURATION_NAME + " configuration file. Using 127.0.0.1:9998 as contact point");
+            return TikaConfiguration.builder()
```

Comment:
If no config, maybe we can fallback to an EmbeddedTikaTextExtractor, with appropriate warnings positioned.

This would provide me, who run a single James server for a few 10s users on a RasberryPy to have to install yet another Java service.

We can:
 - Have an isEmbedded() Method on the config
 - We default saying it's embedded
 - Then we can fallback when provisionning TextExtractor to the Embedded version

What do you think of this?